feat: issue_10

yoshikawa-akr opened the pull request:
# [issue No.10](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#10)

**issue**
問題9に追加で try-catch-finaliy を書いてください

## [CodeSandBox](https://codesandbox.io/s/aged-tdd-6yfcqn) / Latest commit: https://github.com/yoshikawa-akr/js-lessons/pull/10/commits/c00a013b96342f52c636f7e629120a07762f8f9f
## Implementation concerns
Whether the contents of the process to be written in try, catch, and finally are appropriate or not.

yoshikawa-akr committed:
feat: Add try/catch/finally processing to the code for issue 9.

csakyo commented on issue_10/script.js:
```diff
@@ -0,0 +1,58 @@
+async function showList() {
+    startLoading();
+    try {
+        const listData = await getListData();
+        createLists(listData);
```
In this case, only 'getData' should be implemented in the ‘try', because if an error occurs, it is impossible to determine where the error occurred. (I was reviewed like this before!)

yoshikawa-akr replied:
Thanks for your suggestion:sparkles:
It is indeed difficult to identify when an error occurs.
I have modified it to define another function and execute createList there.
https://github.com/yoshikawa-akr/js-lessons/pull/10/commits/059005bc219206e7c56da0639fc5e839b8726f62

csakyo replied:
I think it's good!!:thumbsup:

yoshikawa-akr committed:
refactor: Modification of function processing.

Limit processing in the try block to one process and execute createList in a separate function.

csakyo commented on issue_10/script.js:
```diff
@@ -0,0 +1,61 @@
+
+function getListData() {
+    return new Promise((resolve) => {
+        const lists = [{
+            img: "1.png",
+            alt: "画像1",
+            text: "ブックマーク"
+        }, {
+            img: "2.png",
+            alt: "画像2",
+            text: "メッセージ"
+        }];
```
You're missing an element. You should use this.
```js
[{to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"}, {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}]
```

The results are also different.
You should make this.
```html
<ul>
 <li><a href="/bookmark.html"><img src="1.png" alt="画像1">ブックマーク</a></li>
 <li><a href="/message.html"><img src="2.png" alt="画像2">メッセージ</a></li>
</ul>
```

どの課題からかわかりませんが、問題の内容が違っていますね・・:sob:

yoshikawa-akr replied:
Thank you for pointing that out.
I mistakenly changed from a tag to p tag and deleted the elements in the array as well.
It has been corrected.
https://github.com/yoshikawa-akr/js-lessons/pull/10/commits/c00a013b96342f52c636f7e629120a07762f8f9f

yoshikawa-akr committed:
refactor: Array modification.

Add "to" to lists and specify it in the href attribute of the a tag.
